src/main/webapp/resources/js/pages/analysis/components/AnalysisProvenance.jsx

```diff
@@ -1,12 +1,206 @@
+  grey3,
+  blue2,
+  blue3,
+  blue4,
+  blue5,
+  blue6,
+  blue7,
+  blue8,
+  blue9,
+  blue10
+} from "../../../styles/colors";
+import { getAnalysisProvenanceByFile } from "../../../apis/analysis/analysis";
+import { TabPaneContent } from "../../../components/tabs/TabPaneContent";
+import { AnalysisContext } from "../../../contexts/AnalysisContext";
+import { AnalysisOutputsContext } from "../../../contexts/AnalysisOutputsContext";
+
+const { Panel } = Collapse;
 
 export default function AnalysisProvenance() {
+  const { analysisContext } = useContext(AnalysisContext);
+  const { analysisOutputsContext, getAnalysisOutputs } = useContext(
+    AnalysisOutputsContext
+  );
+  const [provenance, setProvenance] = useState(null);
+  const [toolInfo, setToolInfo] = useState([]);
+  const [currFileName, setCurrFileName] = useState(null);
+
+  const toolPanelMargin = 5;
+  const levelColors = [
+    {
+      0: blue10,
+      1: blue9,
+      2: blue8,
+      3: blue7,
+      4: blue6,
+      5: blue5,
+      6: blue4,
+      7: blue3,
+      8: blue2
+    }
+  ];
+
+  // Gets the analysis outputs on load if they have not
+  // already been loaded
+  useEffect(() => {
+    if (analysisOutputsContext.outputs === null) {
+      getAnalysisOutputs();
+    }
+  }, []);
+
+  // Get the provenance for the file that the user selected
+  function getProvenance(filename) {
+    if (typeof filename !== "undefined") {
+      if (
+        (provenance !== null &&
+          !provenance.data.filename.includes(filename.toString())) ||
+        provenance === null
+      ) {
+        setCurrFileName(filename.toString());
+        getAnalysisProvenanceByFile(
+          analysisContext.analysis.identifier,
+          filename.toString()
+        ).then(data => {
+          setProvenance(data);
+          setToolInfo([]);
+        });
+      }
+    }
+  }
+
+  // Returns a collapse panel for the tool and it's execution parameters.
+  // Parameters are sorted alphabetically by label
+  function displayToolWithExecutionParameters(tool, margin) {
+    let parameters = [];
+    let sortedParameters = tool.executionParameters.sort((a, b) =>
+      a.parameterName > b.parameterName ? 1 : -1
+    );
+
+    for (let param of sortedParameters) {
+      parameters.push(
+        <Descriptions.Item
+          label={param.parameterName}
+          key={`${tool.toolName}-${param.parameterName}`}
+        >
+          {param.parameterValue}
+        </Descriptions.Item>
+      );
+    }
+
+    return (
+      <Collapse
+        style={{
+          border: "none",
+          borderLeft: `8px ${getLevelColor(margin)} solid`,
+          borderRadius: 0,
+          backgroundColor: grey1
+        }}
+        key={`collapse-${tool.toolName}-${Math.random()}`}
+      >
+        <Panel
+          header={tool.toolName}
+          key={tool.toolName}
+          style={{
+            backgroundColor: grey1,
+            marginLeft: `${3 * margin}px`,
+            borderBottom: "0px",
```

Comment:
Just put `0` not `"0px"` in jsx

Reply:
Updated in 7165a33809647dafedbd205e730437decb47dde8